### PR TITLE
[Do not merge] Add section support

### PR DESCRIPTION
**Pull request by feich-ms**
Add section support in lu file. Fixes https://github.com/microsoft/BotFramework-Composer/issues/733

### CODE REVIEW

File: packages/luis/parser/lufile/LUFileLexer.g4 [hunk 1]
```diff
@@ -1,7 +1,7 @@
 lexer grammar LUFileLexer;
 
 @lexer::members {
-  this.ignoreWS = true;             // usually we ignore whitespace, but inside template, whitespace is significant
+  ignoreWS = true;             // usually we ignore whitespace, but inside template, whitespace is significant
```
Comment (boydc2014) [inline packages/luis/parser/lufile/LUFileLexer.g4]: inside intent or inside untterance

File: packages/luis/parser/lufile/LUFileParser.g4 [hunk 2]
```diff
@@ -8,6 +8,7 @@ file
 
 paragraph
     : newline
+    | sectionDefinition
```
Comment (boydc2014) [inline packages/luis/parser/lufile/LUFileParser.g4]: sectionDefintiion now contains intentDefinition? right?

Comment (feich-ms) [reply packages/luis/parser/lufile/LUFileParser.g4]: not exactly. Intent can be outside section like 
# section
## intent1InSection
- xxx

# standaloneIntent
- xxx

File: packages/luis/src/parser/converters/luistoluconverter.js [hunk 3]
```diff
@@ -48,7 +48,7 @@ module.exports = {
                     }
                     fileContent += NEWLINE + NEWLINE;
                 }
-                fileContent += '## ' + intent.intent.name + NEWLINE;
+                fileContent += '# ' + intent.intent.name + NEWLINE;
```
Comment (vishwacsena) [inline packages/luis/src/parser/converters/luistoluconverter.js]: We should chat. This to me seems like we will break back compat.

**Comment by feich-ms** [comment]
The test coverage drop mainly comes from newly added CRUD operations in luParser.js. Will add more tests soon.

**Comment by vishwacsena** [comment]
We can review this once 
- Tests are reverting to keep the `## intent` so we do not break back-compat
- Remove luParser.js CRUD operations from this PR. Make a separate PR with this + tests.

@feich-ms please update this PR once the changes are in.

**Comment by feich-ms** [comment]
@vishwacsena 
1. I have reverted all the test cases that changed "##" to "#"..
2. The CRUD changes are also removed.
3. The changes in json files of test cases are just related to content order, not content itself
4. if '> !# @enableSections = true' is not added in the header, below content will be identified as two intents(AA and BB, AA is an intent without utterances)
   #AA 
   ##BB
   \- hello world

   if '> !# @enableSections = true' is added in the header, above content will be identified as section(AA) and sub intent(BB)
6. For the test coverage drop, I have no idea why it drops so much. I looked at the generated report, it is mainly caused by parseFileContents.js. But based on the report, my newly added functions or contents in parseFileContents.js are all covered by my newly added tests. so maybe @munozemilio can help to take a look?

**Comment by vishwacsena** [comment]
@feich-ms We discussed this in BF-CLI review meeting today and decided to take these + other related changes for composer into a composer branch once we are locked down for ignite release (hopefully this week). So marking this as Do not merge. The only changes we want to take to master for now are related to Ignite deliverables.

**Comment by feich-ms** [comment]
@vishwacsena, Ok, please let me know when the composer branch is ready. Thanks.

**Comment by feich-ms** [comment]
Close this PR first, since all the composer related changes including section and section CRUD support are in this branch https://github.com/microsoft/botframework-cli/tree/composer now. When it is decided to go into the master, we can create a new PR then.

**Pull request closed without merging**